src/main/java/cubicchunks/server/CubeIO.java

```diff
@@ -542,7 +542,7 @@ private static NBTTagCompound writeCubeToNbt(final Cube cube) {
 		// tile entities
 		NBTTagList nbtTileEntities = new NBTTagList();
 		nbt.setTag("TileEntities", nbtTileEntities);
-		for (TileEntity blockEntity : cube.getTileEntityMap().values()) {
+		for (TileEntity blockEntity : cube.getTileEntityMap()) {
```

Comment:
I wholeheartedly agree and I honestly have no idea why that changed. I do not recall changing that line or anything in CubeIO for that matter. Honestly, I should not have started working on this today, I'm sorry for causing this much work.